apps/src/code-studio/components/ShareAllowedDialog.jsx

```diff
@@ -13,6 +15,31 @@ import { showPublishDialog } from '../../templates/publishDialog/publishDialogRe
 import PublishDialog from '../../templates/publishDialog/PublishDialog';
 import { createHiddenPrintWindow } from '@cdo/apps/utils';
 import i18n from '@cdo/locale';
+import firehoseClient from '@cdo/apps/lib/util/firehose';
+
+function recordShare(type) {
+  if (!window.dashboard) {
+    return;
+  }
+
+  firehoseClient.putRecord({
+    study: 'finish-dialog-share',
+    study_group: 'v1',
+    event: 'project-share',
+    project_id: dashboard.project && dashboard.project.getCurrentId(),
```

Comment:
nit: it should be possible to import project.js directly, could be worth a quick try to avoid adding globals to a new file.

Reply:
I was following the pattern in https://github.com/code-dot-org/code-dot-org/pull/21541/files#diff-2c78d143e20bb7302d6491188c882258R95.  Should that file be changed as well?  Thanks!

Reply:
Oops, I assumed for some reason that the dashboard.project was necessary in that file, because I saw it was already being used in there. That one should have been changed over to import project.js as well. Shame on me! 😊 You're welcome to change that one too, or feel free to just fix it in this file and I can fix the other one in a separate PR.

Reply:
Synced with @davidsbailey offline — there's a circular dependency when including `project.js` -> `header.js` -> `ShareDialog.jsx`.  Solving the circular dependency is out of the scope of this PR... leaving as-is for now.